server/channels/api4/job.go

```diff
@@ -147,23 +148,48 @@ func getJobs(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	jobType := r.URL.Query().Get("job_type")
 	var validJobTypes []string
-	for _, jobType := range model.AllJobTypes {
+
+	if jobType != "" {
 		hasPermission, permissionRequired := c.App.SessionHasPermissionToReadJob(*c.AppContext.Session(), jobType)
 		if permissionRequired == nil {
-			c.Logger.Warn("The job types of a job you are trying to retrieve does not contain permissions", mlog.String("jobType", jobType))
-			continue
+			c.Err = model.NewAppError("getJobsByType", "api.job.retrieve.nopermissions", nil, "", http.StatusBadRequest)
+			return
+		}
+		if !hasPermission {
+			c.SetPermissionError(permissionRequired)
+			return
 		}
-		if hasPermission {
-			validJobTypes = append(validJobTypes, jobType)
+		validJobTypes = append(validJobTypes, jobType)
+	} else {
+		for _, jType := range model.AllJobTypes {
+			hasPermission, permissionRequired := c.App.SessionHasPermissionToReadJob(*c.AppContext.Session(), jType)
+			if permissionRequired == nil {
+				c.Logger.Warn("The job types of a job you are trying to retrieve does not contain permissions", mlog.String("jobType", jType))
+				continue
+			}
+			if hasPermission {
```

Comment:
Do we need to do anything when `!hasPermission`?

Reply:
No we just discard it from the jobs that we are allowed to fetch. This slice just contains jobs that we have permission to view